src/LttngHelpers.c

```diff
@@ -796,20 +913,320 @@ static int EventProbeFilter(
             }
         }
 
-        for (struct lttng_bytecode_runtime *pRuntime = cds_list_entry(
-                 lttngh_RCU_DEREFERENCE(pEvent->bytecode_runtime_head.next), __typeof__(*pRuntime), node);
+#if lttngh_UST_VER >= 213
+
+        struct lttng_ust_probe_ctx probeCtx = {
+            .struct_size = sizeof(probeCtx),
+            .ip = pCallerIp };
+        enableRecorder =
+            !pEvent->eval_filter ||
+            pEvent->run_filter(pEvent, stackData, &probeCtx, NULL) == LTTNG_UST_EVENT_FILTER_ACCEPT;
+        if (enableRecorder &&
+            pEvent->type == LTTNG_UST_EVENT_TYPE_NOTIFIER)
+        {
+            struct lttng_ust_event_notifier* pEventNotifier = (struct lttng_ust_event_notifier*)pEvent->child;
+            struct lttng_ust_notification_ctx context = {
+                .struct_size = sizeof(struct lttng_ust_notification_ctx),
+                .eval_capture = CMM_ACCESS_ONCE(pEventNotifier->eval_capture) };
+            pEventNotifier->notification_send(pEventNotifier, stackData, &probeCtx, &context);
+        }
+
+#else // lttngh_UST_VER
+
+        enableRecorder = pEvent->has_enablers_without_bytecode;
+        struct lttng_bytecode_runtime* pRuntime;
+        for (pRuntime = cds_list_entry(
+                 lttng_ust_rcu_dereference(pEvent->bytecode_runtime_head.next), __typeof__(*pRuntime), node);
              &pRuntime->node != &pEvent->bytecode_runtime_head;
              pRuntime = cds_list_entry(
-                 lttngh_RCU_DEREFERENCE(pRuntime->node.next), __typeof__(*pRuntime), node))
+                 lttng_ust_rcu_dereference(pRuntime->node.next), __typeof__(*pRuntime), node))
         {
             if (caa_unlikely(pRuntime->filter(pRuntime, stackData) & LTTNG_FILTER_RECORD_FLAG))
             {
-                filterRecord = 1;
+                enableRecorder = 1;
             }
         }
+
+#endif // lttngh_UST_VER
+    }
+
+    return enableRecorder;
+}
+
+struct lttngh_EventProbeContext
+{
+    unsigned maxAlign;
+    unsigned cbData;
+    unsigned char* pbTranscodeScratch;
+    unsigned cbTranscodeScratch;
+};
+
+static int EventProbeComputeSizes(
+    struct lttngh_EventProbeContext* pContext,
+    struct lttngh_DataDesc* pDataDesc,
+    unsigned cDataDesc)
+    lttng_ust_notrace;
```

Comment:
Extracted this subroutine from the `EventProbe` function. Not changed at all -- just needed to be able to invoke this chunk of code from both branches of an ifdef.